packages/core-web/src/index.js

```diff
@@ -52,38 +52,76 @@ function detectAndApplyFixedHeaderStyles() {
 
+  const resizeHeader = () => {
+    const newHeaderHeight = headerSelector.height();
+    const sheets = document.styleSheets;
+    for (let i = 0; i < sheets.length; i += 1) {
+      const rules = sheets[i].cssRules;
+      // eslint-disable-next-line lodash/prefer-get
+      if (rules && rules[0] && rules[0].selectorText) {
+        switch (rules[0].selectorText) {
+        case '.fixed-header-padding':
+          sheets[i].deleteRule(0);
+          sheets[i].insertRule(`.fixed-header-padding { padding-top: ${newHeaderHeight}px !important }`);
+          break;
+        case 'span.anchor':
+          rules[0].style.top = `calc(-${newHeaderHeight}px - ${bufferHeight}rem)`;
+          break;
+        case 'span.card-container::before':
+          rules[0].style.marginTop = `calc(-${newHeaderHeight}px - ${bufferHeight}rem)`;
+          rules[0].style.height = `calc(${newHeaderHeight}px + ${bufferHeight}rem)`;
+          break;
+        case '.nav-menu-open':
+          rules[0].style.maxHeight = `calc(100% - ${newHeaderHeight}px + 50px)`;
+          break;
+        default:
+          break;
+        }
+      }
+    }
+  };
+
+  const toggleHeaderOverflow = () => {
+    const headerMaxHeight = headerSelector.css('max-height');
+    if (headerMaxHeight === '100%') {
+      headerSelector.css('overflow', '');
+      resizeHeader();
+    }
+  };
+
   const addResizeHeaderListener = () => {
     const resizeObserver = new ResizeObserver(() => {
-      const newHeaderHeight = headerSelector.height();
-      const sheets = document.styleSheets;
-      for (let i = 0; i < sheets.length; i += 1) {
-        const rules = sheets[i].cssRules;
-        // eslint-disable-next-line lodash/prefer-get
-        if (rules && rules[0] && rules[0].selectorText) {
-          switch (rules[0].selectorText) {
-          case '.fixed-header-padding':
-            sheets[i].deleteRule(0);
-            sheets[i].insertRule(`.fixed-header-padding { padding-top: ${newHeaderHeight}px !important }`);
-            break;
-          case 'span.anchor':
-            rules[0].style.top = `calc(-${newHeaderHeight}px - ${bufferHeight}rem)`;
-            break;
-          case 'span.card-container::before':
-            rules[0].style.marginTop = `calc(-${newHeaderHeight}px - ${bufferHeight}rem)`;
-            rules[0].style.height = `calc(${newHeaderHeight}px + ${bufferHeight}rem)`;
-            break;
-          case '.nav-menu-open':
-            rules[0].style.maxHeight = `calc(100% - ${newHeaderHeight}px + 50px)`;
-            break;
-          default:
-            break;
-          }
-        }
+      const headerMaxHeight = headerSelector.css('max-height');
+      // set overflow to hidden if max-height is not 100%
+      if (headerMaxHeight !== '100%') {
+        headerSelector.css('overflow', 'hidden');
+        // re-check after 0.6s to account for transition time
+        setTimeout(toggleHeaderOverflow, 600);
```

Comment:
Generally not a good idea to use `setTimeout` unless a delay / timer is explicitly needed, as the timing of events (here being the transition finishing and resize handler) is not guaranteed. (especially across browsers / devices)

You could take a look at the https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/transitionend_event (and run) here instead to detect whether it is transitioning / finished transitioning

Reply:
Thanks for the suggestion @ang-zeyu :) Was not aware of this transition event. This makes it much better. Have removed the `setTimeout` and replaced it with `transitionend` listener.